src/main/java/com/amazon/opendistroforelasticsearch/sql/executor/format/Protocol.java

```diff
@@ -203,4 +278,79 @@ private JSONArray dataEntry(Row dataRow, Schema schema) {
         }
         return entry;
     }
-}
+
+    public void generateCursorId() {
+        // TODO: only to be used for generating cursor from first page
+        // for subsequent pages the cursorType and cursor should be set from
+        switch(cursorType) {
+            case DEFAULT:
+                int pages_left = pagesLeft();
+                if (options.get("scrollId") != null && pages_left > 0) {
+                    JSONObject cursorJson = new JSONObject();
+                    cursorJson.put("type", cursorType.name());
+                    cursorJson.put("schema", getSchemaAsJson());
+                    cursorJson.put("scrollId", options.get("scrollId"));
+                    cursorJson.put("left", pagesLeft());
+                    setIndexNameInCursor(cursorJson);
+                    setFieldAliasMapInCursor(cursorJson);
+                    cursor = encodeCursorContext(cursorJson);
+
+                } else {
+                    // explicitly setting this to null to avoid any ambiguity
+                    cursor = null;
+                }
+                options.remove("scrollId");
+                options.remove("fetch_size");
+                break;
+            case AGGREGATION:
+                throw new ElasticsearchException("Cursor not yet supported for GROUP BY queries");
```

Comment:
FeatureNotSupportedExecption?

Reply:
Removed this logic as part of refactoring.